backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinService.kt

```diff
@@ -104,6 +116,104 @@ class IsinService(
         }.receive()
     }
 
+    @Suppress("ReturnCount")
+    override suspend fun tryCreateVaccinationAndDose(
+        vaccination: StoreVaccinationRequestDto,
+        patient: PatientDtoOut
+    ): Boolean {
+        if (patient.isinId == null) {
+            logger.info("No ISIN ID provided for patient ${patient.id}. Skipping vaccination creating in ISIN.")
+            return false
+        }
+        if (vaccination.vaccineExpiration == null) {
+            logger.info("No vaccine expiration provided for vaccination ${vaccination.vaccinationId}. Skipping vaccination creating in ISIN.")
+            return false
+        }
+
+        val vaccinationExpirationInstant = vaccination.vaccineExpiration.atTime(LocalTime.MIDNIGHT).atZone(ZoneId.systemDefault()).toInstant();
+
+        val indication = if (patient.indication == null || patient.indication.isBlank())
+            "J01"
+        else
+            patient.indication
+
+        return runCatching {
+            val isinVaccination = createVaccination(
+                IsinVaccinationCreateOrUpdateDtoIn(
+                    id = null,
+                    pacientId = patient.isinId,
+                    typOckovaniKod = "CO19",
+                    indikace = listOf(indication),
+                    indikaceJina = configuration.indikaceJina
+                )
+            )
+
+            if (isinVaccination.id == null) {
+                throw NoSuchFieldException("We expect ISIN return non null vaccination id from create vaccination api call")
+            }
+
+            logger.debug("Creating vaccination in ISIN was successful for patient with ISIN ID ${patient.isinId}.")
+            logger.debug("Data obtained from ISIN: $isinVaccination")
+
+            val dose = createVaccinationDose(
+                IsinVaccinationDoseCreateOrUpdateDtoIn(
+                    id = null,
+                    vakcinaceId = isinVaccination.id,
+                    ockovaciLatkaKod = configuration.ockovaciLatkaKod,
+                    datumVakcinace = vaccination.vaccinatedOn,
+                    typVykonuKod = null,
```

Comment:
to je podle me 1 -> jakoze prvni davka

​/api​/v1​/ciselniky​/Nacti​/TypyVykonu